nixos/modules/services/databases/postgresql.nix

```diff
@@ -531,7 +657,14 @@ in
                 sleep 0.1
             done
 
-            if test -e "${cfg.dataDir}/.first_startup"; then
+            if [ -e "${cfg.dataDir}/.first_startup" ]; then
+              if [ -e "${cfg.dataDir}/.post_upgrade" ]; then
+                '' + optionalString cfg.analyzeAfterUpgrade ''
```

Comment:
This looks odd to me (terminating the multiline string at a different indentation level etc). What's wrong with `${optionalString cfg.analyzeAfterUpgrade ''\n`?